### PR TITLE
Add Orchard generators

### PR DESCRIPTION
**Pull request by therealyingtong**

### CODE REVIEW

File: orchard_generators.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+SPENDING_KEY_BASE = group_hash(b'z.cash:Orchard', b'G')
+
+# https://zips.z.cash/protocol/nu5.pdf#commitmentsandnullifiers
+NULLIFIER_K_BASE = group_hash(b'z.cash:Orchard', b'K')
+
+# https://zips.z.cash/protocol/nu5.pdf#concretehomomorphiccommit
+VALUE_COMMITMENT_VALUE_BASE = group_hash(b'z.cash:Orchard-cv', b'v')
+VALUE_COMMITMENT_RANDOMNESS_BASE = group_hash(b'z.cash:Orchard-cv', b'r')
+
+# Used in SinsemillaCommit (https://zips.z.cash/protocol/nu5.pdf#sinsemillacommitments)
```
Comment (str4d) [inline orchard_generators.py:19]:
```suggestion
# Used in SinsemillaCommit (https://zips.z.cash/protocol/nu5.pdf#concretesinsemillacommit)
```

File: orchard_generators.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+NOTE_COMMITMENT_BASE = group_hash(b'z.cash:Orchard-NoteCommit-r', b'')
+NOTE_COMMITMENT_Q = group_hash(b'z.cash:Orchard-NoteCommit-M', b'')
```
Comment (str4d) [inline orchard_generators.py:21]: The domain separator for SinsemillaHashToPoint is used as the message for GroupHash when deriving Q:
```suggestion
NOTE_COMMITMENT_Q = group_hash(b'z.cash:SinsemillaQ', b'z.cash:Orchard-NoteCommit-M')
```

File: orchard_generators.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+# Used in SinsemillaShortCommit (https://zips.z.cash/protocol/nu5.pdf#sinsemillacommitments)
```
Comment (str4d) [inline orchard_generators.py:23]:
```suggestion
# Used in SinsemillaShortCommit (https://zips.z.cash/protocol/nu5.pdf#concretesinsemillacommit)
```

File: orchard_generators.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+IVK_COMMITMENT_BASE = group_hash(b'z.cash:Orchard-CommitIvk-r', b'')
+IVK_COMMITMENT_Q = group_hash(b'z.cash:Orchard-CommitIvk-M', b'')
```
Comment (str4d) [inline orchard_generators.py:25]:
```suggestion
IVK_COMMITMENT_Q = group_hash(b'z.cash:SinsemillaQ', b'z.cash:Orchard-CommitIvk-M')
```

File: orchard_generators.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+# Used in SinsemillaHash (https://zips.z.cash/protocol/nu5.pdf#orchardmerklecrh)
+MERKLE_CRH_Q = group_hash(b'z.cash:Orchard-MerkleCRH', b'')
```
Comment (str4d) [inline orchard_generators.py:28]:
```suggestion
MERKLE_CRH_Q = group_hash(b'z.cash:SinsemillaQ', b'z.cash:Orchard-MerkleCRH')
```

**Review by str4d — CHANGES_REQUESTED** [review]

**Comment by daira** [comment]
This is included in #28.

**Comment by daira** [comment]
Closing in favour of #21 (to be merged first) and then #28.

**Pull request closed without merging**